src/games/strategy/util/SHA256Crypt.java

```diff
@@ -0,0 +1,23 @@
+package games.strategy.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Replacement for MD5Crypt
+ * using  SHA256 instead
+ */
+public class SHA256Crypt {
+  /**
+   * Returns the SHA256-Hash of the given String
+   */
+  public static String crypt(String text) throws UnsupportedEncodingException, NoSuchAlgorithmException{
```

Comment:
Well the Exceptions are only thrown if the Strings of the algorithm type or the charset is changed to something invalid